src/github.com/mozilla-services/pushgo/simplepush/structs.go

```diff
@@ -29,48 +28,32 @@ type PushCommand struct {
 }
 
 type PushWS struct {
-	uaidLock sync.RWMutex
-	uaid     string // Hex-encoded client ID; not normalized
-	Socket   Socket // Remote connection
+	uaid   string // Hex-encoded client ID; not normalized
+	Socket Socket // Remote connection
 	Store
 	Logger    *SimpleLogger
 	Metrics   *Metrics
 	Born      time.Time
-	closeLock sync.RWMutex
-	closed    bool
+	closeOnce Once
 }
 
 func (ws *PushWS) UAID() (uaid string) {
-	ws.uaidLock.RLock()
-	uaid = ws.uaid
-	ws.uaidLock.RUnlock()
-	return
+	return ws.uaid
 }
 
 func (ws *PushWS) SetUAID(uaid string) {
-	ws.uaidLock.Lock()
 	ws.uaid = uaid
-	ws.uaidLock.Unlock()
 }
 
 func (ws *PushWS) IsClosed() (closed bool) {
```

Comment:
Yes, the better way is for RemoveClient to take two args. The uaid *and* the client, and RemoveClient will only remove the uaid if the client objects match. This way if a new one is added, old RemoveClient calls will not remove the new uaid/client.